### PR TITLE
Enable wide chars for all Windows

### PR DESCRIPTION
**Pull request by kmilos**
This is platform specific, not compiler specific (i.e. it is also implemented by libstdc++ and libc++ on Win32 such as MinGW).

### CODE REVIEW

**Comment by kmilos** [comment]
Hm, the CI for ming32/mingw64 fails, presumably because your cross-compile toolchain on Ubuntu doesn't have wchar enabled? The `7.3-posix` in

```
heif_file.cc:70:149: error: no matching function for call to ‘std::basic_ifstream<char>::basic_ifstream(const wchar_t*, const openmode&)’
   auto input_stream_istr = std::unique_ptr<std::istream>(new std::ifstream(convert_utf8_path_to_utf16(input_filename).c_str(), std::ios_base::binary));
                                                                                                                                                     ^
In file included from heif_file.cc:23:0:
/usr/lib/gcc/x86_64-w64-mingw32/7.3-posix/include/c++/fstream:521:7: ...
```

makes me think this is not a real Win32 toolchain...

[There is no problem building this on Msys2 w/ "native" toolchains.](https://github.com/msys2/MINGW-packages/pull/10228/checks)

**Comment by kmilos** [comment]
Yep, it seems [there is also a `7.3-win32` toolchain](https://packages.ubuntu.com/bionic/amd64/gcc-mingw-w64-x86-64/filelist) available, but no idea how prefer the use of that one...

**Comment by kmilos** [comment]
Scratch that, posix/win32 is only about threading model...

Wide char support is available only from g++-mingw-w64-x86-64_8.3.0, so a CI update is more than welcome.

File: examples/heif_thumbnailer.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@
 #  include "encoder_png.h"
 #endif
 
-#if defined(_MSC_VER)
+#if defined(__MINGW32__) || defined(__MINGW64__) || defined(_MSC_VER)
```
Comment (lazka) [inline examples/heif_thumbnailer.cc:45]: I think you can just `#ifdef _WIN32`

Comment (kmilos) [reply examples/heif_thumbnailer.cc:45]: Thanks, I was thinking about that too, but then wasn't be sure one wants this for Cygwin/MSYS as well (perhaps multibyte UTF8 just works there)?

In either case, still stuck w/ the older libstdc++ problem...

W/ the separate MSVC/MINGW, one could potentially also check for `__GLIBCXX__` version...

Comment (lazka) [reply examples/heif_thumbnailer.cc:45]: I don't think that's relevant on cygwin, locale encoded filenames should just work there.

**Comment by lazka** [comment]
I've created #570

**Comment by fancycode** [comment]
@kmilos could you please check if your changes work with the updated toolchain from #570?

**Comment by kmilos** [comment]
Well, [they certainly work on the latest MSYS2 toolchains](https://github.com/msys2/MINGW-packages/pull/10228), but I can later perhaps try and rebase on top of #570 if you want to check the ubuntu-20.04 image specifically?

**Comment by fancycode** [comment]
> [...] I can later perhaps try and rebase on top of #570 if you want to check the ubuntu-20.04 image specifically?

That would be perfect, thanks!

**Comment by fancycode** [comment]
Nice, thanks for confirming.

**Pull request merged**